### PR TITLE
fix: CoinMarket Service was fixed

### PR DESCRIPTION
**Pull request by ahmedrns**
fix: boa-stat API was updated

### CODE REVIEW

File: src/modules/storage/LedgerStorage.ts [hunk 1]
```diff
@@ -3501,11 +3501,14 @@ export class LedgerStorage extends Storages {
...
-            FROM
-                blocks;`;
+        const sql = `
+                SELECT max(height) as height,
+	                (SELECT SUM(total_reward) as total_reward from accounts) as total_reward,
+                    (SELECT SUM(total_frozen) as total_frozen from accounts) as total_frozen,
+                    (SELECT count(*) from transactions) as transactions,
+                    (SELECT count(DISTINCT address) from validators) as validators
+                    FROM
+                    blocks;`;
```
Comment (TrustHenry) [inline src/modules/storage/LedgerStorage.ts]: It seems like a query that lot of costs.
Can you reduce the DB load?
Is aggregate data not available?

File: src/Stoa.ts [hunk 2]
```diff
@@ -2081,9 +2082,8 @@ class Stoa extends WebService {
                     if (changes)
                         logger.info(
                             `Saved a pre-image utxo : ${pre_image.utxo.toString().substr(0, 18)}, ` +
-                                `hash : ${pre_image.hash.toString().substr(0, 18)}, pre-image height : ${
-                                    pre_image.height
-                                }`,
+                            `hash : ${pre_image.hash.toString().substr(0, 18)}, pre-image height : ${pre_image.height
```
Comment (TrustHenry) [inline src/Stoa.ts]: Is there a reason why you use `substr`?
Why don't we loged the entire hash?

Comment (ahmedrns) [reply src/Stoa.ts]: This was used before, in fact entire hash can be logged

Comment (ahmedrns) [reply src/Stoa.ts]: Entire hash is now being logged

File: src/modules/storage/LedgerStorage.ts [hunk 3]
```diff
@@ -3501,11 +3501,16 @@ export class LedgerStorage extends Storages {
...
-             (SELECT count(*) from transactions) as transactions,
-             (SELECT count(*) from validators) as validators
-            FROM
-                blocks;`;
+        const sql = `
+                SELECT MAX(B.height) as height,
+                    SUM(B.tx_count) as transactions,
+	                SUM(BS.total_reward) as total_reward,
+                    (SELECT SUM(total_frozen) as total_frozen from accounts) as total_frozen,
+                    (SELECT COUNT(DISTINCT address) from validators) as validators
```
Comment (TrustHenry) [inline src/modules/storage/LedgerStorage.ts:3509]: I don't think this value is right.
You must count the number of valid validators for the corresponding block height. Right?

Comment (ahmedrns) [reply src/modules/storage/LedgerStorage.ts:3509]: Active validators are yet to be added, these are the total number of the validators of the system, active validators will be fetched on every block height

Comment (TrustHenry) [reply src/modules/storage/LedgerStorage.ts:3509]: This part needs to be corrected later.

**Review by TrustHenry — APPROVED** [review]

**Pull request merged**